edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Test/EdgeletFixture.cs

```diff
@@ -29,19 +29,22 @@ class ServiceHost
 
             ServiceHost()
             {
-                this.webHostTask = BuildWebHost(new string[0], DefaultPort).RunAsync(this.cancellationTokenSource.Token);
+                this.webHostTask = CreateHostBuilder(new string[0], DefaultPort).Build().RunAsync(this.cancellationTokenSource.Token);
                 this.Url = $"http://localhost:{DefaultPort}";
             }
 
             public static ServiceHost Instance { get; } = new ServiceHost();
 
             public string Url { get; }
 
-            static IWebHost BuildWebHost(string[] args, int port) =>
-                WebHost.CreateDefaultBuilder(args)
-                    .UseUrls($"http://*:{port}")
-                    .UseStartup<Startup>()
-                    .Build();
+            static IHostBuilder CreateHostBuilder(string[] args, int port) =>
```

Comment:
nit: call Build in this method and return IHost

Reply:
I think Andrew follows what did in aspnet core template, https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#hostbuilder-replaces-webhostbuilder.

Reply:
I made this change to netstandard 2.1, but it causes a problem building. I investigated for a while then started trying to find a workaround. I've altered the buildBranch script so that we don't ```dotnet build``` the tests to an output folder, but rather ```dotnet publish``` them to an output folder. I don't know why we weren't doing this initially.

There are still issues running the tests on linux. Still working on fixing them. Windows is working fine.

Reply:
I figured out how to fix the build with netstandard without needing to switch build to publish.

Reply:
I think it is ok to follow the template here. If we change here we should also change the other test fixtures, TestAnalyzer, and TRC